### PR TITLE
Add test class for shmem library.

### PR DESCRIPTION
**Pull request by hectorespert**
Fixes #

**Description of the Change**

Add test class for shmem library.

Copy shmem test program code from lib/shmem_test.cpp to tests/unit-tests/lib/test_shmem.cpp and run it as a unit test.

**Alternate Designs**

**Release Notes**

### CODE REVIEW

File: tests/unit-tests/lib/test_shmem.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
+// This file is part of BOINC.
+// http://boinc.berkeley.edu
+// Copyright (C) 2020 University of California
```
Comment (AenBleidd) [inline tests/unit-tests/lib/test_shmem.cpp:3]: Please change to `2021`

File: tests/unit-tests/lib/test_shmem.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+//
+// BOINC is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+// See the GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with BOINC.  If not, see <http://www.gnu.org/licenses/>.
+
+#ifndef _WIN32
```
Comment (AenBleidd) [inline tests/unit-tests/lib/test_shmem.cpp:18]: Why this can't be tested on Windows too?

**Review by AenBleidd — CHANGES_REQUESTED** [review]

File: tests/unit-tests/lib/test_shmem.cpp (hunk 2 above)
Comment (hectorespert) [reply tests/unit-tests/lib/test_shmem.cpp:18]: I can't test it in windows. These are the errors that the GitHub actions shows: https://github.com/BOINC/boinc/runs/2107473806

```
(ClCompile target) -> 
         D:\a\boinc\boinc\lib\shmem.h(37,20): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [D:\a\boinc\boinc\win_build\unittests_vs2019.vcxproj]
         D:\a\boinc\boinc\lib\shmem.h(37,8): error C2146: syntax error: missing ';' before identifier 'create_shmem' [D:\a\boinc\boinc\win_build\unittests_vs2019.vcxproj]
         D:\a\boinc\boinc\lib\shmem.h(40,20): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [D:\a\boinc\boinc\win_build\unittests_vs2019.vcxproj]
         D:\a\boinc\boinc\lib\shmem.h(40,1): error C2086: 'int HANDLE': redefinition [D:\a\boinc\boinc\win_build\unittests_vs2019.vcxproj]
         D:\a\boinc\boinc\lib\shmem.h(40,8): error C2146: syntax error: missing ';' before identifier 'attach_shmem' [D:\a\boinc\boinc\win_build\unittests_vs2019.vcxproj]
         D:\a\boinc\boinc\lib\shmem.h(41,25): error C2146: syntax error: missing ')' before identifier 'hSharedMem' [D:\a\boinc\boinc\win_build\unittests_vs2019.vcxproj]
        ```

Comment (AenBleidd) [reply tests/unit-tests/lib/test_shmem.cpp:18]: Try to put this to the `include` section:

```
#if defined(_WIN32)
#include "boinc_win.h"
#endif
```

**Comment by AenBleidd** [comment]
Closed in favor of #4294 that contains fixes for Window

**Pull request closed without merging**